lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -339,103 +357,88 @@ void ThreadManager::Impl::addWorker(size_t value) {
     idMap_.insert(std::pair<const Thread::id_t, shared_ptr<Thread> >((*ix)->getId(), *ix));
```

Comment:
Okay actually that comment was wrong.  maxMonitor_ will wake under two conditions - one is that there's enough room to add a task.  The other was not coded and that is during stop.  If there was a thread in add() and someone called stop() it needs to signal maxMonitor_ and if the thread manager is no longer started we need to throw the illegal state exception since it's bieng stopped.